doc/command_line/field_processing.md

```diff
@@ -0,0 +1,126 @@
+## Field Processing
+
+Ruby supports <i>field processing</i>.
+
+This means that when certain command-line options are given,
+the invoked Ruby program can process input line-by-line.
+
+### About the Examples
+
+Examples here assume that this code (creating file `desiderata.txt`)
+has been executed:
+
+```
+text = <<EOT
+Go placidly amid the noise and the haste,
+and remember what peace there may be in silence.
+As far as possible, without surrender,
+be on good terms with all persons.
+EOT
+File.write('desiderata.txt', text)
+```
```

Comment:
IMO this is more confusing than it has to. We can just say that `desiderata.txt` contains the contents.

Reply:
As in some of the other docs, this gives the user a cut-and-paste route to actually running the examples.  But I'll replace with a remark if you prefer.

Reply:
Isn't this more work though? The user has to copy and paste this script and run this script. If we directly have the script, then user can just copy and paste the text.

Reply:
FIxed.